sdk/formrecognizer/Azure.AI.FormRecognizer/perf/Tests/RecognizeCustomFormsWithoutLabelsTest.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.AI.FormRecognizer.Tests;
+using Azure.AI.FormRecognizer.Training;
+using Azure.Test.Perf;
+
+namespace Azure.AI.FormRecognizer.Perf
+{
+    public class RecognizeCustomFormsWithoutLabelsTest : FormRecognizerTest<PerfOptions>
+    {
+        private readonly FormRecognizerClient _client;
+
+        private readonly FormTrainingClient _trainingClient;
+
+        private string _modelId;
+
+        public RecognizeCustomFormsWithoutLabelsTest(PerfOptions options) : base(options)
+        {
+            _client = new FormRecognizerClient(new Uri(Endpoint), new AzureKeyCredential(ApiKey));
+            _trainingClient = new FormTrainingClient(new Uri(Endpoint), new AzureKeyCredential(ApiKey));
+        }
+
+        public override async Task SetupAsync()
+        {
+            var trainingClient = new FormTrainingClient(new Uri(Endpoint), new AzureKeyCredential(ApiKey));
+            var op = await trainingClient.StartTrainingAsync(new Uri(BlobContainerSasUrl), useTrainingLabels: false);
+
+            CustomFormModel model = await op.WaitForCompletionAsync();
+            _modelId = model.ModelId;
+        }
+
+        public override async Task CleanupAsync()
+        {
+            await _trainingClient.DeleteModelAsync(_modelId);
+        }
+
+        public override async Task RunAsync(CancellationToken cancellationToken)
+        {
+            var op = await _client.StartRecognizeCustomFormsFromUriAsync(_modelId, CreateUri(TestFile.Form1));
+            await op.WaitForCompletionAsync();
+        }
+
+        public override void Run(CancellationToken cancellationToken)
+        {
+            var op = _client.StartRecognizeCustomFormsFromUri(_modelId, CreateUri(TestFile.Form1));
+            op.WaitForCompletionAsync().GetAwaiter().GetResult();
+        }
```

Comment:
We're forced to provide a synchronous implementation for our tests, since it's an abstract method in the grandparent class. `WaitForCompletionAsync` does not have a sync counterpart, so I wonder what's the best way to test it.

How would customers make a complete synchronous call to the service? Maybe using `UpdateStatus` directly while the operation hasn't completed? It seems that testing LROs is still a difficult open discussion that needs to be addressed by those working in the perf framework.

Reply:
yeah, not sure there is value of having this. Same as in the meeting.
@witemple-msft for context and guidance

Reply:
JS didn't seem to have a sync run method when I checked. Python seems to be able to poll synchronously.

Reply:
Sorry, to be more clear:
>How would customers make a complete synchronous call to the service? Maybe using UpdateStatus directly while the operation hasn't completed

Yeah, that will be the only way. To do manual polling using UpdateStatus

>It seems that testing LROs is still a difficult open discussion that needs to be addressed by those working in the perf framework.

I was just wondering if it makes sense to add this project/tests knowing that all of them are LRO and there isn't a clear story on what we actually achieve by measuring the LRO behavior

Reply:
Discussed with Mike and we agreed on having throughput tests for LROs for the time being, such as this one. To have a useful measure, we would need to span multiple LROs in parallel. He suggested running it with 1000 threads, but he also raised a concern we didn't consider about the service costs we would end up with.

Until we have a clear plan on reducing costs, I'm limiting the FR perf tests to 1 thread per run to avoid triggering unnecessary expensive runs by accident:
https://github.com/Azure/azure-sdk-for-net/blob/d3c8d5bbcf370d1143b25d9cc3d7296580f17c9f/sdk/formrecognizer/Azure.AI.FormRecognizer/perf/FormRecognizerTest.cs#L19-L22